lib/data/editor/editor_history.dart

```diff
@@ -115,11 +116,13 @@ class EditorHistoryItem {
     this.offset,
     this.page,
     this.quillChange,
+    this.colorChange,
   })  : assert(type != EditorHistoryItemType.move || offset != null, 'Offset must be provided for move'),
         assert(type != EditorHistoryItemType.deletePage || page != null, 'Page must be provided for deletePage'),
         assert(type != EditorHistoryItemType.insertPage || page != null, 'Page must be provided for insertPage'),
         assert(type != EditorHistoryItemType.quillChange || quillChange != null, 'Quill change must be provided for quillChange'),
-        assert(type != EditorHistoryItemType.quillUndoneChange || quillChange != null, 'Quill change must be provided for quillUndoneChange');
+        assert(type != EditorHistoryItemType.quillUndoneChange || quillChange != null, 'Quill change must be provided for quillUndoneChange'),
+        assert(type != EditorHistoryItemType.changeColor || colorChange != null || colorChange!.length == strokes.length, 'ColorChange must be provided and must have the same length as strokes for changeColor');
```

Comment:
I don't think this binary logic would work

```suggestion
        assert(type != EditorHistoryItemType.changeColor || colorChange?.length == strokes.length, 'ColorChange must be provided and must have the same length as strokes for changeColor');
```